### ISSUE TITLE
allow admin to change user email address

### ISSUE THREAD

**Issue by acodispo** [delight-im/PHP-Auth#69]
Am I correct in understanding that there's currently no way for an admin user to change the email address of a particular user (and thereby send them a confirmation email)?

I've asked a related question on #53, about whether the email address of an impersonated user can be changed.

Thanks!

**Comment by ocram** [comment]
Well, it really depends on what you want to do exactly. Let’s check if what you want might already be there:

 1. If a user didn’t get their confirmation email, you can just use the – usually public – interface for requesting another email, which the user themself would use as well. Just enter their email address there. Of course, to do that, you must have implemented this feature.
 1. If a user can’t or doesn’t want to log in to their account in order to change their email address, you can impersonate them, which means that you skip the password check as an admin and just log in as the user directly. Then, again, you can use the normal interface that the user themself would use to change their email address. Again, you must obviously have implemented that feature. If changing one’s email address causes a confirmation email to be sent with your service (which it should), the email will be sent in this case as well. So the user receives a link at their new address which they can use to complete the address change.

Does this include what you need? Or is it something else?

**Comment by acodispo** [comment]
@ocram Thanks! I've implemented solution 1, for circumstances where the user didn't receive their confirmation. The situation here is that (for whatever reason) the user doesn't want to log in to change their email address. I will use solution 2 (and thanks for your answer over in #53).

It's up to you, but I might suggest leaving this issue open as a feature request, i.e. that an admin user have the ability to directly modify a particular user's email address (targeting the user via ID or current email address).

**Comment by ocram** [comment]
You’re right, we should leave this open and implement this as another feature in the `Administration` interface in the future.

The advantage will be that you will no longer have to switch from the admin context to the user context by impersonating the user, requiring that you switch back to the admin context afterwards. With the new feature, you could stay in the admin context. Furthermore, we could allow admins to change email addresses *without* any confirmation as well.

Thanks!

**Comment by acodispo** [comment]
@ocram Great, thanks!
